relation/index.js

```diff
@@ -108,7 +110,25 @@ module.exports = yeoman.generators.Base.extend({
           }
           return m;
         },
-        validate: validateName
+        validate: function(value) {
+
+          var done = this.async();
+
+          if (validateName(value) !== true) {
+            done(validateName(value));
```

Comment:
Ooops, I did not realize `done` expects an error string. In that case you should definitely save the result of `validateName` to a variable.